include/swift/AST/Decl.h

```diff
@@ -2973,6 +2973,14 @@ class ValueDecl : public Decl {
   /// \c \@usableFromInline, \c \@inlinalbe, and \c \@_alwaysEmitIntoClient
   bool isUsableFromInline() const;
 
+  /// Treat as public and allow skipping type check if the following conditions
+  /// are met:
+  /// - This decl has a package access level.
+  /// - Has a @usableFromInline (or other inlinable) attribute
+  /// - And is defined in a module built from a public or private
+  ///   interface that does not contain package-name.
+  bool treatAsPublic() const;
```

Comment:
This name reads a bit too generically to me - if I hadn't read the documentation I wouldn't know what it represents at its use sites. What about something like `hasEffectivelyPublicPackageAccess()`?

Reply:
Sg, probably `hasEffectivelyPublicAccess()`.

Reply:
I think it should mention `package` somehow because it's only meant to be called when you've already determined that the decl has `package` level access. I would find it confusing if `hasEffectivelyPublicAccess()` returns `false` for a decl that is already marked `public` for example.

Reply:
How about `isPackageEffectivelyPublic`?

Reply:
That's better, though it feels like it's missing an `And` or a `But`, e.g. `isPackageButEffectivelyPublic`.

Reply:
To be more concrete, it could be `isInterfacePackageEffectivelyPublic`.